### PR TITLE
MINIFICPP-733 - Redirect librdkafka log to proper Logger

### PR DESCRIPTION
**Pull request by bakaid**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by bakaid** [comment]
I am not sure what would be the better solution: currently I create loggers for KafkaConnections. It would be possible (if not elegant) to pass the PublishKafka logger for every connection and use that for logging. That way all logs would logged under the PublishKafka and not the KafkaConnection class.

**Comment by bakaid** [comment]
@arpadboda Loggers are members of KafkaConnections and exist unchanged during the entire lifetime of the KafkaConnections. rd_kafka_t handles - to which loggers are associated in the static map - have lifetimes lower than or equal to KafkaConnections (the rd_kafka_t handle can be changed in the KafkaConnection with setConnection).
This means that there is an external (external from smart pointers) guarantee that the same logger will exist during the entire lifetime of the rd_kafka_t handle, and a shared_ptr held by the static map is not necessary to extend the logger lifetime to the lifetime of the rd_kafka_t handle.
While it is not necessary to have a shared_ptr, why would it be harmful to have one?
If everything worked as intended, it would make no difference. However, if there is a bug now or in later iterations of the code, whereby we fail to remove the entry from the static map when the rd_kafka_t handle is freed, then when the KafkaConnection is finally destroyed, its logger will still be held until the static map is freed (i.e. the entire lifetime of the application).
In the case of a logger this would not cause a great issue except for the small amount of leaked memory, but in more complicated shared_ptr configurations it might cause more serious errors.
To summarize, I have arrived to this design decision because of two factors:
 - the logger lifetime is guaranteed in other ways than holding a shared_ptr
 - I am dislike storing shared_ptrs in static variables, and I like to fail safe in these situations, even though an error in the current case does not pose a great threat
